### DISCUSSION TITLE
Simplify if-else conditions to null coalesce / ternary

### DISCUSSION THREAD

**Discussion by HenkPoley** [PHP-CS-Fixer/PHP-CS-Fixer#7392] [Feature ideas]
New rule proposal: transforming simple if/else-based assignments into null coalesce operator.

Code example:
```php
if (!isset($a)) {
    $b = $some_default;
} else {
    $b = $a;
}
```

Expected output:
```php
$b = $a ?? $some_default;
```

Side note: It might also be clearer for single-line if-not-else statements to simply flip both arms and remove the not.

**Comment by SpacePossum** [comment]
Nice idea, we might even try a `if-to-ternary` rule, for example:

```
    if ($a) {
        $b = $some_default;
    } else {
        $b = $a;
    }

//

$b = $a ? $some_default : $a;
```

if we would run that one before TernaryToNullCoalescingFixer than your case would be fixed as well :)
